src/renderer/components/ft-shaka-video-player/player-components/AutoplayToggle.js

```diff
@@ -0,0 +1,77 @@
+import shaka from 'shaka-player'
+
+import i18n from '../../../i18n/index'
+
+export class AutoplayToggle extends shaka.ui.Element {
+  /**
+   * @param {EventTarget} events
+   * @param {HTMLElement} parent
+   * @param {shaka.ui.Controls} controls
+   */
+  constructor(autoplayEnabled, events, parent, controls) {
+    super(parent, controls)
+
+    /** @private */
+    this.button_ = document.createElement('button')
+    this.button_.classList.add('autoplay-toggle')
+    this.button_.classList.add('shaka-tooltip')
+
+    /** @private */
+    this.icon_ = document.createElement('i')
+    this.icon_.classList.add('material-icons-round')
+    this.icon_.textContent = 'pause_circle'
+
+    this.button_.appendChild(this.icon_)
+
+    const label = document.createElement('label')
+    label.classList.add('shaka-overflow-button-label')
+    label.classList.add('shaka-overflow-menu-only')
+
+    /** @private */
+    this.nameSpan_ = document.createElement('span')
+    label.appendChild(this.nameSpan_)
+
+    /** @private */
+    this.currentState_ = document.createElement('span')
+    this.currentState_.classList.add('shaka-current-selection-span')
+    label.appendChild(this.currentState_)
+
+    this.button_.appendChild(label)
+
+    this.parent.appendChild(this.button_)
+
+    /** @private */
+    this.autoplayEnabled_ = autoplayEnabled
+
+    // listeners
+
+    this.eventManager.listen(this.button_, 'click', () => {
+      events.dispatchEvent(new CustomEvent('toggleAutoplay', {
+        detail: !this.autoplayEnabled_
+      }))
```

Comment:
As you don't seem to be using the payload in the event listener on the other side, we can send the event without the payload.
```suggestion
      events.dispatchEvent(new CustomEvent('toggleAutoplay'))
```

Reply:
That's being used on Line 55, no?
https://github.com/FreeTubeApp/FreeTube/blob/4317f3e0dc0c0ade04335461246adad1595f0f6c/src/renderer/components/ft-shaka-video-player/player-components/AutoplayToggle.js#L55

Reply:
Whoops, yes you are correct, I only looked in the player file itself (maybe I shouldn't be reviewing pull requests on my phone :P ).